src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let obj = Object.assign({}, state);
+
+  for (const action of actions) {
+    obj = Object.assign({}, obj, action.extraData);
+
+    if (action.type === 'addProperties') {
+      res.push(obj);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const removeElement of action.keysToRemove) {
+        delete obj[removeElement];
+      }
+
+      res.push(obj);
+    }
+
+    if (action.type === 'clear') {
+      obj = {};
+      res.push(obj);
+    }
```

Comment:
so after condition you can use `push`
```
    res.push({ ... obj });
```

Reply:
<img width="766" alt="image" src="https://github.com/mate-academy/js_stateful-clones/assets/70220667/a2d481c0-ae7f-4a5a-bd62-ea4dafee6407">
still not fixed

Reply:
<img width="723" alt="image" src="https://github.com/mate-academy/js_stateful-clones/assets/70220667/5077fd13-17fa-4e22-901b-99a6da06c1c1">
you should fix it everywhere